colearn_keras/keras_learner.py

```diff
@@ -100,11 +134,24 @@ def mli_propose_weights(self) -> Weights:
         - Current model is reverted to original state after training
         :return: Weights after training
         """
+        if self.diff_priv_config is not None:
+            epsilon_after_training = self.get_privacy_budget()
+            if epsilon_after_training > self.diff_priv_budget.target_epsilon:
+                return Weights(
```

Comment:
This part might needs to be discussed:
What to do when we cannot train any more due to privacy violation?
 - the actual spent privacy is still < than the budget
 - training another round would violate privacy, thus it is prohibited
 - returning old weights does not makes sense (computational waste and it might seem that weight is returned even when it is a privacy violation)
 
 #### returning `weights=None` might be too general? Or is it enough?
 - should we handle in the  `weight=None` in the `mli_test_weights`? to return -1/NaN vote/test score and False for vote
 - it should be also handles in the orchestrator, but if there are multiple reasons for returning weights=None, how can the orchestrator know if it is due to differential privacy?

Reply:
After our conversation:
 - always return a valid weight object
 - when further training is not possible, return the last weight & an error code in the training summary